Make it possible to run command line interaction together with UI.

kramlie opened the pull request:
This is just a port of what has already been done for GTK and Qt.

kramlie commented:
@jofemodo: I hope this can be considered for Zynthian!

This goes together with https://github.com/zynthian/zynthian-ui/pull/507.

jofemodo commented:
Hi @kramlie !

This needs to be rebased against the asyncli branch to work with the latest version.

Thanks

kramlie commented:
Hey @jofemodo!

Looks like there is a conflicting implementation of the same thing in the asyncli branch, but it doesn't work correctly. I will need a bit of time to figure that out and fix it. Unless you are planning to look at it yourself?

The symptom is that when using the `jalv -s` option to enable the GUI, the command line interface stops working. In my original implementation the command line and GUI could respond at the same time.

kramlie committed:
jalv_console: Fix indentation.

Signed-off-by: Kristian Amlie <kristian@amlie.name>

kramlie committed:
jalv_console: Run command line interaction together with UI.

After 099171489e0fe it is possible to start the command line thread
separately from the UI, it just didn't check for this condition
upon starting. By splitting the two checks we allow for both the
command line and graphical interface to be accessible at the same
time.

Signed-off-by: Kristian Amlie <kristian@amlie.name>

kramlie commented:
Alright, @jofemodo, luckily this turned out to be a pretty small fix. All the pieces were there, it just didn't check the flags correctly when starting. I have now updated so it can be merged to the asyncli branch!

jofemodo commented:
Merged!
